src/github_utils.py

```diff
@@ -19,111 +20,107 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content: str) -> List[dict]:
+    """Parse diff content with precise line tracking"""
+    hunks = []
     current_file = None
-    line_number = None
+    new_line = None
     
     for line in diff_content.split('\n'):
-        # Reset tracking for new files
         if line.startswith('diff --git'):
-            current_file = line.split(' b/')[-1].strip()
-            valid_lines[current_file] = set()
-            line_number = None  # Reset for new file
-            continue
-            
-        # Parse hunk header
-        if line.startswith('@@'):
+            current_file = line.split(' b/')[1].split()[0]
+            new_line = None
+        elif line.startswith('@@'):
             try:
```

Comment:
Improved file parsing logic to get only the file path.
```suggestion
|
```
N/A